src/android/Dgram.java

```diff
@@ -75,198 +48,191 @@ public void run() {
             DatagramPacket datagramPacket = new DatagramPacket(data, data.length);
             while (true) {
                 try {
-                    if (this.m_datagramSocket.isClosed()) {
+                    if (this.datagramSocket.isClosed()) {
```

Comment:
there is a danger here of the datagramSocket being null if someone calls listen before open.

Reply:
i see that there is still protection in the execute for not `listening` or `sending` if the socket is null